CMakeLists.txt

```diff
@@ -38,7 +38,8 @@
 # To check:
 # (From Daniel Stenberg) The cmake build selected to run gcc with -fPIC on my box while the plain configure script did not.
 # (From Daniel Stenberg) The gcc command line use neither -g nor any -O options. As a developer, I also treasure our configure scripts's --enable-debug option that sets a long range of "picky" compiler options.
-cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.0...3.16 FATAL_ERROR)
```

Comment:
Why did you change this to put in a max version

Reply:
It sets all policies introduced up to 3.16 to new, for all versions that support this syntax. Older versions will just ignore that.
https://cmake.org/cmake/help/v3.16/command/cmake_minimum_required.html

Reply:
I'm not sure if I understand this. Is it that if a later version for example 3.17 introduces some new policies then this warns about them?

Reply:
Policies warn only if code is hit where a policy would make a difference in behavior _and_ the policy is not explicitely set to either OLD or NEW. So if there is code that would change behavior in 3.18 and is build with that version then there would be warnings. Since 3.16 is the latest released version I don't think it's a good idea to set this to any higher value. So far only 2 policy warnings were triggered (CMP0054 and CMP0075), so most others were just silent because no affected code was used.

Reply:
Ok. My concern was that it functions on all versions from 3.0 on, which I guess it does. As you can tell I don't know a lot about cmake.

Reply:
Yes, this syntax has been chosen because older versions will silently ignore it.